src/query/shorthand.ts

```diff
@@ -277,3 +282,205 @@ export function fieldDef(encQ: EncodingQuery,
   }
   return fieldAndParams;
 }
+
+const CHANNEL_INDEX = toMap(CHANNELS);
+const AGGREGATE_OP_INDEX = toMap(AGGREGATE_OPS);
+const SINGLE_TIMEUNIT_INDEX = toMap(SINGLE_TIMEUNITS);
+const MULTI_TIMEUNIT_INDEX = toMap(MULTI_TIMEUNITS);
+
+export function parse(shorthand: string): SpecQuery {
+  // TODO(https://github.com/uwdata/compassql/issues/259):
+  // Do not split directly, but use an upgraded version of `getClosingBraceIndex()`
+  let splitShorthand = shorthand.split('|');
+
+  let specQ: SpecQuery = {mark: splitShorthand[0], encodings: [] as EncodingQuery[]};
+
+  for (let i = 1; i < splitShorthand.length; i++) {
+    let part = splitShorthand[i];
+    const splitPart = splitWithTail(part, ':', 1);
+    const splitPartKey = splitPart[0];
+    const splitPartValue = splitPart[1];
+
+    if (CHANNEL_INDEX[splitPartKey] || splitPartKey === '?') {
+      const encQ = shorthandParser.encoding(splitPartKey, splitPartValue);
+      specQ.encodings.push(encQ);
+      continue;
+    }
+
+    if (splitPartKey === 'calculate') {
+      let transformQ: TransformQuery = specQ.transform || {};
+
+      let calculate = [];
+      let formulas = splitPartValue.split(',');
```

Comment:
Actually, fixing the shorthand method for calculate to be correct (#260) would take < 10 mins.  
How about you send another PR to fix that and update this one (parse) to be correct -- instead of leaving it like this.